core/src/main/scala/org/http4s/Uri.scala

```diff
@@ -279,6 +277,104 @@ object Uri {
     }
   }
 
+  final class Path private (
+      val segments: List[String],
```

Comment:
I think you fixed some tests by abandoning `Chain`, but it seems like it was a good model.  `Path` appends will be common.  Maybe the constant factors are small enough `List` is better?

Reply:
We could use Vector, but Scala < 2.13.2 incurs a memory penalty on small vectors. Most Paths are pretty short < 10 segments, so I dont think that the cost of append might be that expensive

Reply:
And random access will be rare.  As long as we're not concatenating them often, sounds good to me.